### PR TITLE
Fixes #16878: Fix generation of parameters and conditions from cfengine files

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16878

bug corrections : parameters partly done, and conditions fix, parenthesis conditions a bit better

### CODE REVIEW

File: rudder-lang/tools/tester.sh [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
-cfjson_tester=/opt/rudder/share/rudder-lang/tools/cfjson_tester
-technique_path=/var/rudder/configuration-repository/techniques/ncf_techniques/1.0/technique.cf
+dir=$PWD/tests/test_files/tester/$name
+cfjson_tester=$PWD/tools/cfjson_tester
 
 mkdir -p $dir/
 
 # Take original technique an make a json
-$cfjson_tester ncf-to-json $technique_path $dir/technique.json
+$cfjson_tester ncf-to-json $dir/technique.cf $dir/technique.json /tools/rudderc-dev.conf
```
Comment (peckpeck) [inline rudder-lang/tools/tester.sh]: there should be no / at the beginning of tools/ it would imply that you put it in / which is misleading

Comment (peckpeck) [reply rudder-lang/tools/tester.sh]: either use ./ or nothing

**Comment by gpoblon** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
Test failing

**Comment by peckpeck** [comment]
@gpoblon there are conflits, you must rebase

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Comment by gpoblon** [comment]
rebased, ready to merge

**Comment by amousset** [comment]
@gpoblon Tests are failing

**Comment by gpoblon** [comment]
Yes they are until the next enum PR goes in, @peckpeck is working on it. The oslib.rl does not match the new enum system yet, hence the failing tests

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**